src/login/logind.c

```diff
@@ -444,74 +444,80 @@ static int parse_fdname(const char *fdname, char **session_id, dev_t *dev) {
         return 0;
 }
 
-static int manager_attach_fds(Manager *m) {
-        _cleanup_strv_free_ char **fdnames = NULL;
-        int n, i, fd;
+static void attach_fd(Manager *m, const char *fdname, int fd) {
+        _cleanup_free_ char *id = NULL, *message = NULL;
+        dev_t dev;
+        struct stat st;
+        SessionDevice *sd;
+        Session *s;
+        int r;
 
-        /* Upon restart, PID1 will send us back all fds of session devices
-         * that we previously opened. Each file descriptor is associated
-         * with a given session. The session ids are passed through FDNAMES. */
+        r = parse_fdname(fdname, &id, &dev);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to parse fd name %s: %m", fdname);
+                goto close;
```

Comment:
maybe not call this label "close", given that on gcc you can take pointers of labels, and hence it is not clear whether you mean the close function or the label then...

Reply:
this remains unfixed